libs/features/personalization/personalization.js

```diff
@@ -271,30 +272,91 @@ function parsePlaceholders(placeholders, config, selectedVariantName = '') {
   return config;
 }
 
-function getPersonalizationVariant(manifestPath, variantNames = [], variantLabel = null) {
+export const loadEntitlements = (() => {
+  let entitlements;
+  return (async () => {
+    if (!entitlements) {
+      const { default: getUserEntitlements } = await import('../../blocks/global-navigation/utilities/getUserEntitlements.js');
+      await loadIms();
+      entitlements = getUserEntitlements();
+    }
+    return entitlements;
+  });
+})();
```

Comment:
Strong pattern, but hard to grasp if someone isn't familiar with it.

I'd suggest something that feels more familiar:
```suggestion
async function fetchEntitlements() {
    const { default: getUserEntitlements } = await import('../../blocks/global-navigation/utilities/getUserEntitlements.js');
    await loadIms();
    return getUserEntitlements();
}
let entitlements;
export const loadEntitlements = () => {
    if (!entitlements) entitlements = fetchEntitlements();
    return entitlements;
};
```

Reply:
We use the pattern in a few places, and in general prefer for vars to be encapsulated.  I do like the idea of breaking out `fetchEntitlements` so will change that.

I think it's better that people learn this pattern - even if it's harder to grasp.